src/main/java/be/syntra/devshop/DevshopBack/controllers/ProductController.java

```diff
@@ -4,43 +4,52 @@
 import be.syntra.devshop.DevshopBack.models.CategoryList;
 import be.syntra.devshop.DevshopBack.models.ProductDto;
 import be.syntra.devshop.DevshopBack.models.ProductList;
+import be.syntra.devshop.DevshopBack.models.SearchModelDto;
 import be.syntra.devshop.DevshopBack.services.CategoryService;
 import be.syntra.devshop.DevshopBack.services.ProductService;
+import be.syntra.devshop.DevshopBack.services.SearchService;
+import be.syntra.devshop.DevshopBack.services.utilities.ProductMapperUtility;
+import be.syntra.devshop.DevshopBack.services.utilities.SearchModelMapperUtility;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
 
+@Slf4j
 @RestController
 @RequestMapping("/products")
 public class ProductController {
 
     private final ProductService productService;
     private final CategoryService categoryService;
+    private final SearchService searchService;
+    private final SearchModelMapperUtility searchModelMapperUtility;
+    private final ProductMapperUtility productMapperUtility;
 
     @Autowired
     public ProductController(
             ProductService productService,
-            CategoryService categoryService
+            CategoryService categoryService,
+            SearchService searchService,
+            SearchModelMapperUtility searchModelMapperUtility,
+            ProductMapperUtility productMapperUtility
     ) {
         this.productService = productService;
         this.categoryService = categoryService;
-    }
-
-    @GetMapping()
-    public ResponseEntity<ProductList> retrieveAllNonArchivedProducts() {
-        return ResponseEntity
-                .status(HttpStatus.OK)
-                .body(productService.findAllByArchivedFalse());
+        this.searchService = searchService;
+        this.searchModelMapperUtility = searchModelMapperUtility;
+        this.productMapperUtility = productMapperUtility;
     }
 
     /*
      *@Returns: 201-created code when our product's successfully saved
      */
     @PostMapping()
     public ResponseEntity<ProductDto> createProduct(@RequestBody ProductDto productDto) {
-        productService.save(productDto);
+        productService.save(productMapperUtility.convertToProduct(productDto));
```

Comment:
this methodbody is the exact same as the updateProduct method. Please extract method

Reply:
fixed, intellij extracted the @RequestBody annotation aswell, removed it this time :-)